include/ccmath/math/nearest/rint.hpp

```diff
@@ -10,7 +10,239 @@
 
 #pragma once
 
+#include <ccmath/internal/support/fenv/fenv_support.hpp>
+#include <ccmath/internal/support/fenv/rounding_mode.hpp>
+#include <ccmath/internal/support/fp/directional_rounding_utils.hpp>
+#include <ccmath/math/compare/isinf.hpp>
+#include <ccmath/math/compare/isnan.hpp>
+#include <type_traits>
+
 namespace ccm
 {
+	/**
+	 * @brief Rounds the floating-point argument num to an integer value (in floating-point format), using the current rounding mode. The library provides
+	 * overloads of std::rint for all cv-unqualified floating-point types as the type of the parameter num.
+	 * @tparam T The type of the number.
+	 * @param num A floating-point value.
+	 * @return If no errors occur, the nearest integer value to num, according to the current rounding mode, is returned.
+	 */
+	template <class T, std::enable_if_t<std::is_floating_point_v<T>, bool> = true>
+	constexpr T rint(T num)
+	{
+		if (ccm::isinf(num) || ccm::isnan(num)) { ccm::support::fenv::raise_except_if_required(FE_INVALID); }
+		constexpr auto rounding_mode{ccm::support::fenv::get_rounding_mode()};
```

Comment:
This function can't have `rounding_mode` as `constexpr` because if the compiler evaluates `get_rounding_mode()` at compile time, it will lock in a specific rounding mode (e.g., round-to-nearest) regardless of the actual rounding mode at runtime. This would lead to incorrect rounding behavior when the function is called at runtime since the rounding mode may differ from the one determined at compile time. So I'd remove the `constexpr` qualifier here.

Reply:
Of course you can change this to `const` to take advantage of constant folding!